### PR TITLE
[SYCL][Doc] Add documentation to use iostream_proxy.hpp instead of <iostream>

### PR DESCRIPTION
**Pull request by raaiq1**
This is to complement this [PR](https://github.com/intel/llvm/pull/6469) which removes \<iostream\> headers to comply with LLVM Coding Standards.

Signed-off-by: Rauf, Rana <rana.rauf@intel.com>

### CODE REVIEW

File: sycl/doc/developer/ContributeToDPCPP.md [hunk 1]
```diff
@@ -31,6 +31,9 @@ For any DPC++-related commit, the `[SYCL]` tag should be present in the
 commit message title. To a reasonable extent, additional tags can be used
 to signify the component changed, e.g.: `[PI]`, `[CUDA]`, `[Doc]`.
 
+## Using \<iostream\> 
+
+According to [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html#include-iostream-is-forbidden), the use `#include <iostream>` is forbidden in library files. Thereby it's suggested to use the `<sycl/detail/iostream_proxy.hpp>` header instead, as it has the functionality of `<iostream>` without it's static constructor.
```
Comment (steffenlarsen) [inline sycl/doc/developer/ContributeToDPCPP.md:36]: Please try and keep with the style of the rest of the document, i.e. try and keep lines under the 80 character cap as much as possible.
Please also add an empty line after this.

I would probably reword the second sentence a little, to something like; "Instead, the `sycl/detail/iostream_proxy.hpp` header offers the functionality of `<iostream>` without its static constructor. This header should be used in place of `<iostream>` in DPC++ headers and runtime library files."

Comment (raaiq1) [reply sycl/doc/developer/ContributeToDPCPP.md:36]: Ohh. My bad, didn't know there was a 80 character cap. Will apply the other suggestions

Comment (steffenlarsen) [reply sycl/doc/developer/ContributeToDPCPP.md]: The section still seems to be on one line.

File: sycl/doc/developer/ContributeToDPCPP.md [hunk 2]
```diff
@@ -33,7 +33,8 @@ to signify the component changed, e.g.: `[PI]`, `[CUDA]`, `[Doc]`.
 
 ## Using \<iostream\> 
 
-According to [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html#include-iostream-is-forbidden), the use `#include <iostream>` is forbidden in library files. Instead, the sycl/detail/iostream_proxy.hpp header offers the functionality of <iostream> without its static constructor. This header should be used in place of <iostream> in DPC++ headers and runtime library files.
+According to [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html#include-iostream-is-forbidden), the use `#include <iostream>` is forbidden in library files. Instead, the sycl/detail/iostream_proxy.hpp header offers the functionality of <iostream> without its static constructor.
```
Comment (steffenlarsen) [inline sycl/doc/developer/ContributeToDPCPP.md]:
```suggestion
According to [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html#include-iostream-is-forbidden),
the use `#include <iostream>` is forbidden in library files. Instead, the
sycl/detail/iostream_proxy.hpp header offers the functionality of <iostream>
without its static constructor.
```

**Review by steffenlarsen — APPROVED** [review]

File: sycl/doc/developer/ContributeToDPCPP.md (hunk 1 above)
Comment (raaiq1) [reply sycl/doc/developer/ContributeToDPCPP.md]: By one line I thought you meant the characters within a sentence for some reason

Comment (steffenlarsen) [reply sycl/doc/developer/ContributeToDPCPP.md]: I could have been clearer too, or just made the change suggestion to begin with. 😄

**Pull request merged**